blog-posts/github-billing-dashboard.md

```diff
@@ -0,0 +1,169 @@
+```
+
+```tsx
+export interface UsageReportCsvEntry {
+	date: string;
+	product: string;
+	repositorySlug: string;
+	quantity: string;
+	unitType: string;
+	pricePerUnit: string;
+	actionsWorkflow: string;
+	notes: string;
+}
+
+export interface UsageReportEntry extends UsageReportCsvEntry {
+	totalPrice: number;
+}
+
+parse<UsageReportCsvEntry>(file, {
+	header: true,
+	skipEmptyLines: true,
+	transformHeader: (header: string): string => {
+		return camalize(header);
+	},
+	complete: (result) => {
+			const githubBillingEntries: UsageReportEntry[] = result.data.map(
+				(dailyEntry) => {
+					return {
+					...dailyEntry,
+					totalPrice: *parseFloat*(dailyEntry.quantity) * *parseFloat*(dailyEntry.pricePerUnit),
+				};
+			}
+		);
+	},
+});
+```
+
+### Recharts
+
+Probably the most obvious feature of the dashboard is the functionality to display the costs in charts. For this, we used the React library Recharts. This library comes with some handy features that were also important for this project. Besides the possibility to choose between different chart types (e.g. Bar or Line), Recharts also leaves a lot of room for customizing the charts. This includes, for example, various stylings, additional elements (e.g. tooltip, legend, ...), and the possibility to simply add features that are not available.
+
+```tsx
+<BarChart data={currentData} margin={{ bottom: 20 }}>
+  <CartesianGrid vertical={false} stroke={"rgba(255, 255, 255, 0.1)"} />
+  <XAxis
+    dataKey={groupedBy === "daily" ? "day" : "week"}
+    tick={{ fill: "white" }}
+    axisLine={false}
+    tickLine={false}
+    minTickGap={10}
+  />
+  <YAxis
+    domain={[0, maxValueOfYAxis]}
+    unit=" $"
+    tickCount={maxValueOfYAxis + 1}
+    tick={{ fill: "white" }}
+    axisLine={false}
+    tickLine={false}
+  />
+</BarChart>
+```
+
+### date-fns
+
+For grouping costs by days, weeks and months we had to work a lot with dates. The library `date-fns` made this work easier for us because it brings important functions like `startOfMonth()`, `getISOWeek()`, or `lightFormat()`, which allows you to customize the date format easily. For example, we used the `lightFormat()` function to display the start or end date of the data set. Due to the detailed documentation, it was very pleasant to work with `date-fns`.
+
+```tsx
+isFirstMonth()
+	? ` (from ${lightFormat(new Date(firstDayOfMonth), "dd.MM.")})`
+	: ""
+isLastMonth()
+	? ` (till ${lightFormat(new Date(lastDayOfMonth), "dd.MM.")})`
+	: ""
+```
+
+### react-medium-image-zoom
+
+React-medium-image-zoom is a small but very useful library to zoom in images that are clicked. This library also leaves many possibilities for customization, for example, the background of the zoomed image and the zoom effect can be customized very easily. For the basic function of react-medium-image-zoom it is sufficient to just wrap an image or div tag with the Zoom tag as shown in the example below. The customization features can be added via props.
```

Comment:
That's a cool effect and easy to use 👍. A small video would help understanding what you mean with `zoom`.
We don't have a video player for local files though. So maybe a good feature to develop after the blog post is merged?


https://user-images.githubusercontent.com/1701755/125790084-03d3d72e-b6f3-42c7-8d72-1afb937b746c.mp4

Reply:
Nice idea 👍